src/IceRpc.Templates/Templates/IceRpc-Slice-Client/.template.config/template.json

```diff
@@ -1,57 +1,147 @@
 {
-    "$schema":"http://json.schemastore.org/template",
-    "author":"ZeroC",
-    "classifications":[
+    "$schema": "http://json.schemastore.org/template",
+    "author": "ZeroC",
+    "classifications": [
         "IceRPC",
         "Client",
         "Slice"
     ],
-    "identity":"IceRpc.Slice.Client.0.1",
-    "name":"IceRPC + Slice Client",
-    "description":"A project template for creating an IceRPC + Slice client console application.",
-    "shortName":"icerpc-slice-client",
-    "tags":{
-        "language":"C#",
-        "type":"project"
+    "identity": "IceRpc.Slice.Client.0.1",
+    "name": "IceRPC + Slice Client",
+    "description": "A project template for creating an IceRPC + Slice client console application.",
+    "shortName": "icerpc-slice-client",
+    "tags": {
+        "language": "C#",
+        "type": "project"
     },
-    "sourceName":"IceRpc-Slice-Client",
-    "preferNameDirectory":true,
-    "defaultName":"Client",
-    "symbols":{
-        "skipRestore":{
-            "type":"parameter",
-            "datatype":"bool",
-            "description":"If specified, skips the automatic restore of the project on create.",
-            "defaultValue":"false"
+    "sourceName": "IceRpc-Slice-Client",
+    "preferNameDirectory": true,
+    "defaultName": "Client",
+    "symbols": {
+        "skipRestore": {
+            "type": "parameter",
+            "datatype": "bool",
+            "description": "If specified, skips the automatic restore of the project on create.",
+            "defaultValue": "false"
+        },
+        "transport": {
+            "type": "parameter",
+            "description": "The transport to use for IceRPC connections.",
+            "datatype": "choice",
+            "choices": [
+                {
+                    "choice": "quic",
+                    "description": "Use QUIC transport"
+                },
+                {
+                    "choice": "slic",
+                    "description": "Use slic transport"
+                }
+            ],
+            "defaultValue": "slic",
+            "replaces": "slic"
         }
     },
-    "sources":[
+    "sources": [
         {
-            "modifiers":[
+            "modifiers": [
                 {
-                    "rename":{
-                        "editorconfig":".editorconfig"
+                    "rename": {
+                        "editorconfig": ".editorconfig"
+                    }
+                },
+                {
+                    "condition": "(transport == 'quic')",
```

Comment:
Could we instead use ifdefs in a common source file?

I think this would be more maintainable. With separate source files for TCP (Slic) and QUIC, it's more likely they'll get out of sync.